packages/gatsby-theme-api-docs/src/utils/render-type-as-link.js

```diff
@@ -2,19 +2,35 @@ import React from 'react';
 import { Link } from '@commercetools-docs/gatsby-theme-docs';
 import { markdownFragmentToReact } from '@commercetools-docs/ui-kit';
 import { locationForType } from '../hooks/use-type-locations';
+import { getPrimitiveTypeByName } from '../components/type/type';
 
-function renderTypeAsLink(apiKey, type, typeLocations, description) {
+function renderTypeAsLink(
+  apiKey,
+  type,
+  typeLocations,
+  description,
+  contentType
+) {
   const typeLocation = locationForType(apiKey, type, typeLocations);
 
-  const originalTypeLocation = typeLocation ? typeLocation.url : '';
+  const originalTypeLocation = typeLocation ? typeLocation.url : undefined;
 
-  return originalTypeLocation ? (
-    <Link href={originalTypeLocation}>{type}</Link>
-  ) : description ? (
-    markdownFragmentToReact(description)
-  ) : (
-    type
-  );
-}
+  let primitiveJsonType;
+  if (
+    !originalTypeLocation &&
+    Array.isArray(contentType) &&
+    contentType.includes('application/json')
+  ) {
+    primitiveJsonType = getPrimitiveTypeByName('application/json', type);
+  }
 
+  if (originalTypeLocation) {
+    return <Link href={originalTypeLocation}>{type}</Link>;
+  } else if (primitiveJsonType) {
+    return markdownFragmentToReact(primitiveJsonType);
```

Comment:
(concerning above suggestion to illustrate the idea)
```suggestion
    return <Markdown.Em>{primitiveJsonType}</Mardown.Em>;
```